src/basic/virt.c

```diff
@@ -796,6 +797,22 @@ Virtualization detect_container(void) {
         if (r < 0)
                 log_debug_errno(r, "Failed to detect cgroup namespace: %m");
 
+        /* Finally, the root pid namespace has an hardcoded inode number of 0xEFFFFFFC since kernel 3.8, so
+         * if all else fails we can check the inode number of our pid namespace and compare it. */
+        pidns_fd = namespace_open_by_type(NAMESPACE_PID);
+        if (pidns_fd < 0)
+                log_debug_errno(pidns_fd, "Failed to open PID namespace, ignoring: %m");
+        else {
+                struct stat st;
+
+                if (fstat(pidns_fd, &st) < 0)
+                        log_debug_errno(errno, "Failed to fstat pid namespace fd, ignoring: %m");
+                else if ((uint64_t) st.st_ino != 0xEFFFFFFC) {
+                        v = VIRTUALIZATION_CONTAINER_OTHER;
+                        goto finish;
+                }
+        }
```

Comment:
could you split this into another function `running_in_pidns()` modelled after `running_in_cgroupns()`